snapshot/localmounter_unix.go

```diff
@@ -38,7 +38,13 @@ func (lm *localMounter) Mount() (string, error) {
 		}
 	}
 
-	dir, err := ioutil.TempDir("", "buildkit-mount")
+	// Ensure buidkit/mounts dir exist
+	// TODO perhaps there might be a better place to do this
+	//   Some feedback/pointers would be appreciated
+	if err := os.MkdirAll("/buidkit/mounts/", 0711); err != nil {
```

Comment:
This is definitely not allowed. All the files should be contained either under buildkit state dir or system temp dir.

Reply:
> This is definitely not allowed. All the files should be contained either under buildkit state dir or system temp dir.

Oh Sorry. My bad. I must have missed something in my investigation. The reason why I modified the code like this was because it seems when docker runs buildkit under `userns_remap` mode. There are some logic that calls chroot to `/var/lib/docker/$user_id:$group_id/`. 

I was able to figure this out using `strace`. Here are some of the logs of folders created.
```
1144867 epoll_pwait(5,  <unfinished ...>
1144276 mkdirat(AT_FDCWD, "/var/lib/docker/1000.997/tmp/buildkit-mount959066558", 0700 <unfinished ...>
1144272 <... fcntl resumed>)            = 0x1 (flags O_WRONLY)
1144272 fcntl(48, F_SETFL, O_WRONLY|O_NONBLOCK <unfinished ...>
```

If this proposed approach is against the design, do you mind letting me know what would be a more appropriate solution? 
From the comment I received from @cpuguy83 here https://github.com/moby/moby/pull/43354#issuecomment-1065954668. I though this was the standard approach that buildkit meant to take

Reply:
> here are some logic that calls chroot to

No, this is daemon code that is not chrooted. Only thing that can be on different root is container itself or the extract helper in moby. You can see the full path in your strace output as well. If the process is chrooted all the paths there would be relative to chroot.

Reply:
yeah you are right. 

thanks, I managed to figure out how to compile a version of `dockerd` binary with this change as well`buildkit`

I think I'll try to figure out the correct path by using `config.Root` which seems to be created here `https://github.com/moby/moby/blob/5263bea70f2a8285242da5758f14ebafb176bfc9/cmd/dockerd/daemon.go#L299`